### PR TITLE
[Docs] Add Tutorials

### PR DESCRIPTION
**Pull request by davidmyriel**
Tutorials

Updated Old Tutorials:
- Quickstart
- Filtering: Clauses and Conditions

Added New Tutorials:
- Filtering: Full Text
- Filtering: Nested Filters
- Multivector Search
- Sparse Vector Search
- Hybrid Search (broken, need help here)
- Multitenancy

Screenshot 2024-09-15 at 7 06 45 PM

### CODE REVIEW

File: src/components/InteractiveTutorial/TutorialSubpages.jsx [hunk 1]
```diff
@@ -13,6 +14,8 @@ export const tutorialIndexPage = Index;
 const tutorialSubPages = [
   ['quickstart', Quickstart],
   ['filtering-clauses', FilteringClauses],
+  ['multivectors', Multivectors],
+  ['multitenancy', Multitenancy]
```
Comment (Anush008) [inline src/components/InteractiveTutorial/TutorialSubpages.jsx:18]:
```suggestion
  ['multitenancy', Multitenancy],
```
To fix the lint failure.

File: src/components/InteractiveTutorial/MdxPages/Multitenancy.mdx [hunk 2]
```diff
@@ -0,0 +1,57 @@
+export const title = 'Multitenancy';
+
+# {title}
+
+To use multivectors, create a collection with the following configuration:
```
Comment (Anush008) [inline src/components/InteractiveTutorial/MdxPages/Multitenancy.mdx:5]:
```suggestion
To try multitenancy, create a collection with the following configuration:
```

Comment (Anush008) [reply src/components/InteractiveTutorial/MdxPages/Multitenancy.mdx:5]: * Multitenancy

File: src/components/InteractiveTutorial/MdxPages/Multivectors.mdx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        "type": "ice giant"
+      }
+    }
+  ]
+}
+```
+
+## Step 3: Query the collection 
+
+To perform a search with multivectors, you can pass multiple query vectors. Qdrant will compare the query vectors against the multivectors stored in each point and return the most similar results based on the comparator defined in the collection (`max_sim`). You can also request the payloads to be returned along with the search results.
```
Comment (Anush008) [inline src/components/InteractiveTutorial/MdxPages/Multivectors.mdx:90]:
```suggestion
To perform a search with multivectors, you can pass multiple query vectors. Qdrant will compare the query vectors against the multivectors stored in each point and return the most similar results based on the comparator defined for the collection (`max_sim`). You can also request the payloads to be returned along with the search results.
```

**Review by Anush008 — APPROVED** [review]
Very nice read.
I've left some nits to consider.

File: src/components/InteractiveTutorial/MdxPages/HybridSearch.mdx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        "size": 4,  
+        "distance": "Cosine"  
+    },
+    "sparse_vectors": {
+        "keywords": {
+            "distance": "Cosine"  // Distance metric for sparse vectors
+        }
+    }
+}
+```
```
Comment (kacperlukawski) [inline src/components/InteractiveTutorial/MdxPages/HybridSearch.mdx:18]:
``````suggestion
```json withRunButton=true
PUT /collections/hybrid_search
{
    "vectors": {
        "size": 4,  
        "distance": "Cosine"  
    },
    "sparse_vectors": {
        "keywords": { }
    }
}
```
``````

There is no need to set the distance for sparse vectors.

File: src/components/InteractiveTutorial/MdxPages/HybridSearch.mdx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            "vector": [0.02, 0.4, 0.5, 0.9],   // Dense vector (4D)
+            "sparse_vectors": {
+                "keywords": {
+                   "indices": [1, 42],    // <┐
+                   "values": [0.22, 0.8]  // <┴─sparse vector
+                }
+            }
+        }
+    ]
+}
```
Comment (kacperlukawski) [inline src/components/InteractiveTutorial/MdxPages/HybridSearch.mdx:35]:
```suggestion
PUT /collections/hybrid_search/points
{
    "points": [
        {
            "id": 1,
            "vector": {
            	"": [0.02, 0.4, 0.5, 0.9],   // Dense unnamed vector (4D)
                "keywords": {
                   "indices": [1, 42],    // <┐
                   "values": [0.22, 0.8]  // <┴─sparse vector
                }
            }
        }
    ]
}
```

The dense vector is unnamed, so an empty string has to be passed as a key.

File: src/components/InteractiveTutorial/MdxPages/HybridSearch.mdx [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        },
+        {
+            "query": [0.01, 0.45, 0.67, 0.89], // <-- dense vector
+            "using": "dense",
+            "limit": 20
+        }
+    ],
+    "query": { "fusion": "rrf" }, // <--- reciprocal rank fusion
+    "limit": 10
+}
```
Comment (kacperlukawski) [inline src/components/InteractiveTutorial/MdxPages/HybridSearch.mdx:58]:
```suggestion
POST /collections/hybrid_search/points/query
{
    "prefetch": [
        {
            "query": { 
                "indices": [1, 42],    // <┐
                "values": [0.22, 0.8]  // <┴─sparse vector
             },
            "using": "keywords",
            "limit": 20
        },
        {
            "query": [0.01, 0.45, 0.67, 0.89], // <-- dense vector
            "using": "",
            "limit": 20
        }
    ],
    "query": { "fusion": "rrf" }, // <--- reciprocal rank fusion
    "limit": 10
}
```

The name of the collection and the names of the vectors (empty string and "keywords") were incorrect.

File: src/components/InteractiveTutorial/MdxPages/SparseVectors.mdx [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+        "distance": "Cosine"  
+    },
+    "sparse_vectors": {
+        "text": {  
+            "index": {
+                "on_disk": false  
+            }
+        }
+    }
+}
```
Comment (kacperlukawski) [inline src/components/InteractiveTutorial/MdxPages/SparseVectors.mdx:27]:
```suggestion
PUT /collections/sparse_charts
{
    "sparse_vectors": {
        "text": {  
            "index": {
                "on_disk": false  
            }
        }
    }
}
```

It isn't required to pass dense vectors configuration if you plan to use only sparse vectors.

**Review by kacperlukawski — DISMISSED** [review]
@davidmyriel I made some changes in the hybrid search tutorial, and suggested a minor change in sparse search. Please let me know if you want me to review all the tutorials end-to-end.

**Review by generall — APPROVED** [review]

**Pull request merged**